xla/service/gpu/ir_emitter_unnested.cc

```diff
@@ -2557,63 +2558,77 @@ static std::optional<GlobalDeviceId> DeviceConstraint(
   return std::nullopt;
 }
 
+absl::StatusOr<bool> HasHostMemorySpace(Shape shape, int index) {
+  int host_memory_space = static_cast<int>(stream_executor::MemoryType::kHost);
+  return shape.tuple_shapes(index).has_layout() &&
+      shape.tuple_shapes(index).layout().memory_space() == host_memory_space;
+}
+
+absl::StatusOr<bool> HasHostMemorySpace(Shape shape) {
+  TF_ASSIGN_OR_RETURN(bool is_dst_host_memory, HasHostMemorySpace(shape, 0));
+  TF_ASSIGN_OR_RETURN(bool is_src_host_memory, HasHostMemorySpace(shape, 1));
+  return is_dst_host_memory || is_src_host_memory;
```

Comment:
Behavior of those two helper functions doesn't quite match the description, if it's specific to Copy, maybe it should have a corresponding name?

Reply:
Also do we always need to cast to `int` to check the memory space? Looks counter-intuitive.

Reply:
> Also do we always need to cast to int to check the memory space? Looks counter-intuitive.
Is it better to cast to int once and pass it to the functions?